src/HTTP.jl

```diff
@@ -1,16 +1,16 @@
 __precompile__(true)
 module HTTP
 
-export Request, Response, FIFOBuffer
```

Comment:
why un-export these? that will be breaking

Reply:
My thinking is that `Request` and `Response` are too generic to be exported into the global namespace. For most users writing `HTTP.Response` will be no great burden, and clearer.
Users who really want `Response` in their application module to mean `HTTP.Response` can do `import HTTP.Response`.

As for `FIFOBuffer`, it seems to me that even if we end up returning `FIFOBuffer` objects, the users should not need to know this and should not need to write `FIFOBuffer` anywhere in their code. They should be able to treat all `::IO` streams equally.

On a related note, the `RequestStack.request` function has a `body` parameter, and a `response_stream` parameter. Setting `body` to an `::IO` stream in the request body being streamed and setting `response_stream` to an `::IO` stream results in the response being streamed. In both cases, the users is free to pass whatever stream type they like. i.e. There is no need for `FIFOBuffer` to be part of the `HTTP` package. It could be split into a seperate package and used with `HTTP` if that's what people want. e.g.
```julia
using FIFOBuffer
using HTTP
io = FIFOBuffer()
HTTP.request("GET", "http://foo.bar"; response_stream=io)
```